tests/stub/summary/test_summary.py

```diff
@@ -247,6 +235,791 @@ def test_multiple_notifications(self):
+            }
+        )
+        self.assertEqual(len(summary.gql_status_objects), 2)
+        if self._server.count_requests("PULL"):
+            self.assert_is_no_data(summary.gql_status_objects[0])
+        else:
+            # For drivers that lazily PULL records.
+            # They cannot know whether there is no data until the user
+            # makes the driver try to pull records.
+            self.assert_is_no_data_unknown_subclass(
+                summary.gql_status_objects[0]
+            )
+        self.assert_is_test_notification_as_gql_status_object(
+            summary.gql_status_objects[1],
+            severity="INFORMATION", parsed_severity="INFORMATION"
+        )
+
+    def test_unknown_severity(self):
+        in_notifications = [
+            self.make_test_notification(severity="FOOBAR")
+        ]
+        summary = self._get_summary(
+            "summary_with_notifications.script",
+            vars_={
+                "#NOTIFICATIONS#": json.dumps(in_notifications)
+            }
+        )
+        self.assertEqual(len(summary.gql_status_objects), 2)
+        if self._server.count_requests("PULL"):
+            self.assert_is_success(summary.gql_status_objects[0])
+        else:
+            # For drivers that lazily PULL records.
+            # They cannot know whether there is no data until the user
+            # makes the driver try to pull records.
+            self.assert_is_no_data_unknown_subclass(
+                summary.gql_status_objects[0]
+            )
+        self.assert_is_test_notification_as_gql_status_object(
+            summary.gql_status_objects[1],
+            severity="FOOBAR", parsed_severity="UNKNOWN"
+        )
+
+    def _test_notification_with_missing_data(
+        self, del_key, severity="WARNING"
+    ):
+        notification = self.make_test_notification(severity=severity)
+        del notification[del_key]
+        in_notifications = [notification]
+        summary = self._get_summary(
+            "summary_with_notifications.script",
+            vars_={
+                "#NOTIFICATIONS#": json.dumps(in_notifications)
+            }
+        )
+
+        self.assertEqual(len(summary.gql_status_objects), 2)
+        if self._server.count_requests("PULL"):
+            if severity == "WARNING":
+                self.assert_is_success(summary.gql_status_objects[1])
+                return_index = 0
+            else:
+                self.assert_is_success(summary.gql_status_objects[0])
+                return_index = 1
+        else:
+            # For drivers that lazily PULL records.
+            # They cannot know whether there is no data until the user
+            # makes the driver try to pull records.
+            self.assert_is_no_data_unknown_subclass(
+                summary.gql_status_objects[0]
+            )
+            return_index = 1
+        return summary.gql_status_objects[return_index]
+
+    def test_notification_with_missing_severity(self):
```

Comment:
Following the recent discussions about the empty strings and `null` values, this test should be updated to also accept a diagnostic record map without the `_severity` entry.